### PR TITLE
interfaces/greengrass-support: add additional "process" flavor for 1.11 update

### PR DESCRIPTION
**Pull request by anonymouse64**
This adds a new attribute to the greengrass-support interface, "flavor", which
indicates what mode of containerization the greengrassd daemon is meant to be
supporting with the plug. With no flavor attribute, or the "container" flavor,
then the old policy is available so as to not break old users of the snap, but
with a new "process" flavor, then a far less privileged version of the interface
is provided, which allows the greengrassd daemon to implement no
containerization and thus the lambdas that are run are not run with the
additional privilege afforded to the original implementation of the interface,
as that would allow lambdas to trivially escape the sandbox.

I have tested this change with a version of the greengrass snap that was provided to me by AWS, but my testing is minimal and just ensures that lambdas start-up, etc. AWS has tested the specific policy included here by not using greengrass-support and adding the policy snippets to the AppArmor profile for their new version of the snap and then re-loading the AppArmor profile. However, they would like to have this available in edge ASAP so they can test it "in the field" with the new snap as well to see if there are any other gotchas or other accesses that they might need before their release. As such, because there is a possibility that this might need to be changed, I've milestoned it for 2.49 (and marked it as blocked until 2.48 is branched) so that 2.48 doesn't get blocked waiting for AWS's testing, but conceivably after it is put onto edge and confirmed to be good, it could be backported to 2.48 to be included in 2.48.1 or something.

(the original version of this PR had a much longer description - it was removed to make this more clear for reviewers but can still be viewed with the edit history function in GitHub)

### CODE REVIEW

File: interfaces/builtin/system_observe.go
Comment (jdstrand) [inline interfaces/builtin/system_observe.go:126]: These are fine for system-observe.

File: interfaces/builtin/system_observe.go
Comment (jdstrand) [inline interfaces/builtin/system_observe.go:129]: This is already allowed in hardware-observe.

Comment (anonymouse64) [inline interfaces/builtin/system_observe.go:129]: note that these accesses are also required by Greengrass, but they don't need to go into the specific greengrass-support interface since greengrass also needs system-observe for the other interfaces, so I figured this was an appropriate place to put these accesses.

**Review by jdstrand — CHANGES_REQUESTED** [review]
This PR (granted, intentionally) breaks previous versions of the snap, but I think it is artificially limiting. I suggest taking a different approach: adjust greengrass-support to be more like docker-support and kubernetes-support and have different modes or flavors. In this manner, the default greengrass-support can keep the same permissions, but if you specify the 'process-mode' flavor, then you can cut over to that. They are then free to create a snap that only does process-mode if they want, leaving the door open for more interesting behaviors like creating a 'container' plug and a 'process-mode' plug in the snap and having users decide (similar to privileged containers in docker).

File: interfaces/builtin/system_observe.go
Comment (anonymouse64) [reply interfaces/builtin/system_observe.go]: ack removed, greengrass can just plug hardware-observe instead

File: interfaces/builtin/greengrass_support.go
Comment (jdstrand) [inline interfaces/builtin/greengrass_support.go:448]: We've typically used the pattern in the interfaces that we don't error on unknown attribute values in this manner. IIRC, this was to reduce issues with snaps that might specify an attribute that is in a future release of snapd when running on an older release (eg, the kubernetes-support interface doesn't error). I'll defer to @mvo5 or @pedronis on this point. If changing, also change in SecCompConnectedPlug and AppArmorConnectedPlug.

**Review by jdstrand — APPROVED** [review]
I like the approach, the policy looks good and I like the testsuite changes. Conditionally approving to get out of your way, but please review my attribute comment, I think we don't want to error on unknown attribute.

**Comment by jdstrand** [comment]
Removing Needs Security Review tag.

File: interfaces/builtin/greengrass_support.go
Comment (anonymouse64) [reply interfaces/builtin/greengrass_support.go:448]: sure I can remove this error, that all makes sense

Comment (anonymouse64) [reply interfaces/builtin/greengrass_support.go]: this is done

**Review by mvo5 — APPROVED** [review]
Thank you

**Comment by pedronis** [comment]
this was merged prematurely

@anonymouse64  I would like to understand more what's the upstream terminology for this and is this a configuration or a build  property? 

flavor: container|process

sounds very vague, especially in the singular

**Pull request merged**
